### PR TITLE
raftstore: add metrics for slow ready handle diagnosis

### PR DESCRIPTION
**Pull request by you06**
### What is changed and how it works?

Issue Number: Close #xxx

What's Changed:

```commit-message
Add more details to help the diagnosis of slow ready handle.
```

[code changes](https://github.com/you06/tikv/compare/v4.0.6...v4.0.6-patch)

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by you06** [comment]
@iosmanthus @cosven PTAL

File: components/batch-system/src/metrics.rs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+
+use lazy_static::lazy_static;
+use prometheus::*;
+
+lazy_static! {
+    pub static ref CHANNEL_FULL_COUNTER_VEC: IntCounterVec =
```
Comment (iosmanthus) [inline components/batch-system/src/metrics.rs:7]: Is this backport from newer version?

Comment (you06) [reply components/batch-system/src/metrics.rs:7]: Yes I made a cherry pick

**Comment by you06** [comment]
@Connor1996 PTAL

File: components/raftstore/src/store/fsm/peer.rs [hunk 2]
```diff
@@ -444,9 +448,16 @@ impl<'a, T: Transport, C: PdClient> PeerFsmDelegate<'a, T, C> {
                 }
                 PeerMsg::Noop => {}
             }
+            RAFT_EVENT_DURATION
+                .with_label_values(&[typ])
```
Comment (Connor1996) [inline components/raftstore/src/store/fsm/peer.rs:452]: please use static metrics

Comment (you06) [reply components/raftstore/src/store/fsm/peer.rs:452]: Do you mean the macro `make_auto_flush_static_metric!`? It's not able to use it in release-4.0...

Comment (Connor1996) [reply components/raftstore/src/store/fsm/peer.rs:452]: `make_static_metrics` should be enough

**Review by Connor1996 — APPROVED** [review]

**Comment by iosmanthus** [comment]
LGTM

**Review by iosmanthus — APPROVED** [review]

**Comment by you06** [comment]
@Connor1996 help me merge this PR?

**Pull request merged**